ui/chapter/Chapter.tsx

```diff
@@ -47,7 +48,12 @@ export default function Chapter({ children, metadata, lang }) {
   const chapter = chapters[metadata.slug]
   const position = metadata.position + 1
   const isEven = position % 2 == 0
-
+  const chapterLessons = keys.filter((ele) =>
+    ele.includes(progress.substring(0, 3))
+  )
+  const isBetweenChapter =
+    progress !== chapterLessons[0] &&
```

Comment:
We could add some logic to remove the continue button when the user is at the very Last lesson as there is nothing really to continue to. While the previous logic caused some progression issues this shouldn't have the same problem.

``` suggestion
progress !== keys[keys.length - 1]
```